extensions/common/context.cc

```diff
@@ -316,11 +317,19 @@ bool extractPeerMetadataFromUpstreamHostMetadata(
         CreateKeyVal(fbb, fbb.CreateString(kCanonicalServiceRevisionLabelName),
                      fbb.CreateString(parts[3])));
   }
+  if (parts.size() >= 5) {
+    // In case newer proxy runs with old control plane, only extract cluster
+    // name if there are the fifth part.
+    cluster_id = fbb.CreateString(parts[4]);
+  }
   auto labels_offset = fbb.CreateVectorOfSortedTables(&labels);
 
   FlatNodeBuilder node(fbb);
   node.add_workload_name(workload_name);
   node.add_namespace_(namespace_);
+  if (!cluster_id.IsNull()) {
+    node.add_cluster_id(cluster_id);
```

Comment:
qq: do we need to default this to something if cluster id is null (like "kubernetes" or "unknown")?

Reply:
Currently, with Envoyfilter config, I think the cluster labels will just be `unknown` when it is unknown, and stats filter will fill in unknown for empty labels by default. For example in one of the test, you can find that source cluster is set to unknown when mx is not happening:
```
  - name: source_cluster
    value: unknown
```

Reply:
ok. as long as stats_filter continues to populate `unknown` for missing values, we should be good.